src/ui/Histogram/Histogram.js

```diff
@@ -0,0 +1,692 @@
+/*
+ * Copyright 2022 EPAM Systems, Inc. (https://www.epam.com/)
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useRef, useState, useEffect } from 'react';
+import TransfFunc from '../../engine/TransFunc';
+
+const DEFAULT_HEIGHT = 220;
+const NEED_TO_DRAW_VERTICAL_MARKS = false;
+
+const Histogram = (props) => {
+  const canvasRef = useRef(null);
+  const canvasHistogram = useRef(null);
+  const [width, setWidth] = useState(0);
+  const [height, setHeight] = useState(DEFAULT_HEIGHT);
+  let m_numColors = 0;
+  let m_histogram = [];
+  let m_peakIndex = -1;
+
+  let m_transfFunc = new TransfFunc();
+  let m_transfFuncCallback = undefined;
+  let m_transfFuncUpdateCallback = undefined;
+
+  useEffect(() => {
+    updateCanvas();
+    window.addEventListener('resize', handleResize, false);
+    setSize();
+  }, []);
+
+  useEffect(() => {
+    updateCanvas();
+    window.removeEventListener('resize', handleResize, false);
```

Comment:
I think we can use our own hook and remove event listener after unmounting. For example, https://usehooks.com/useWindowSize/. You can copy `useWindowSize` code from example.

Reply:
Please, check the updated version!